src/api/user_presence.rs

```diff
@@ -13,28 +13,40 @@
 // limitations under the License.
 
 use crate::clock::ClockInt;
-use crate::ctap::Channel;
 use embedded_time::duration::Milliseconds;
 
 pub enum UserPresenceError {
+    // User explicitly declined user presence check.
     Declined,
+    // User presence check was canceled by User Agent.
     Canceled,
+    // User presence check timed out.
     Timeout,
 }
 
 pub type UserPresenceResult = Result<(), UserPresenceError>;
 
 pub trait UserPresence {
-    /// Called at the beginning of user presence checking process.
-    fn check_init(&mut self, channel: Channel);
+    /// Initializes for a user presence check.
+    ///
+    /// Must eventually be followed by a call to [`Self::check_complete`].
+    fn check_init(&mut self);
 
-    /// Implements a wait for user presence confirmation or rejection.
-    fn wait_with_timeout(
-        &mut self,
-        channel: Channel,
-        timeout: Milliseconds<ClockInt>,
-    ) -> UserPresenceResult;
+    /// Waits until user presence is confirmed, rejected, or the given timeout expires.
+    ///
+    /// Must be called between calls to [`Self::check_init`] and [`Self::check_complete`].
+    ///
+    /// # Errors
```

Comment:
nit: I would not repeat the errors here since we have them specified in the enum already. You can remove this section.

Reply:
Done